here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -114,14 +125,66 @@ private Optional<String> getSourceIdFromFeature(NakshaProperties properties) {
     }
   }
 
-  public static Optional<ITagQuery> mapIntoTagOperation(PQuery propertyOperation) {
+  private static boolean isFullyConvertedToITagQuery(IPropertyQuery propertyQuery) {
```

Comment:
A bit misleading name here IMO - we don't do anything with iTagQuery - we just check if property query is"singular"

Reply:
The name makes sense only as the function is called in specific places after 

```
final Optional<ITagQuery> tagComponent = transformPropertyOperation(propertyComponent);
if (tagComponent.isPresent())
```

It indicates that the property query (clause/fragment) has already been converted into tag query (clause/fragment) successfully (and in case of PAnd checking also the size to make sure nothing remains). If it's "fully converted", we can remove the property query (clause/fragment).

But I'm open to new name suggestions :b